ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/linear_optimisation/algorithms/fillers/MaxMinRelativeMarginFiller.java

```diff
@@ -83,14 +68,47 @@ private void buildMinimumRelativeMarginVariable(LinearProblem linearProblem) {
         }
     }
 
+    /**
+     * Build the  miminum relative margin sign binary variable, P.
+     * P represents the sign of the minimum margin.
+     */
+    private void buildMinimumRelativeMarginSignBinaryVariable(LinearProblem linearProblem) {
+        linearProblem.addMinimumRelativeMarginSignBinaryVariable();
+    }
+
     /**
      * Define the minimum relative margin (like absolute margin but by dividing by sum of PTDFs)
      */
     private void buildMinimumRelativeMarginConstraints(LinearProblem linearProblem) {
         MPVariable minRelMarginVariable = linearProblem.getMinimumRelativeMarginVariable();
+        MPVariable minRelMarginSignBinaryVariable = linearProblem.getMinimumRelativeMarginSignBinaryVariable();
+        MPVariable minimumMarginVariable = linearProblem.getMinimumMarginVariable();
+
+        if (minimumMarginVariable == null) {
+            throw new FaraoException("Minimum margin variable has not yet been created");
+        }
         if (minRelMarginVariable == null) {
             throw new FaraoException("Minimum relative margin variable has not yet been created");
         }
+        if (minRelMarginSignBinaryVariable == null)  {
+            throw new FaraoException("Minimum relative margin sign binary variable has not yet been created");
+        }
+        double maxPositiveRelativeRam = highestThreshold / ptdfSumLowerBound;
+        double maxNegativeRelativeRam = 5 * maxPositiveRelativeRam;
+        double maxNegativeRam = 5 * highestThreshold;
+        // Minimum Margin is negative or null
+        minimumMarginVariable.setUb(.0);
```

Comment:
update minimum margin variable in a separate method for readability (like what was done in the method you removed: updateMinimumNegativeMarginDefinitionAndCost(LinearProblem linearProblem))